### PR TITLE
osd: Use specializations, typedefs instead

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

File: src/osd/OSD.h [hunk 1]
```diff
@@ -774,7 +774,7 @@ class OSDService {
 
 private:
   /// throttle promotion attempts
-  std::atomic_uint promote_probability_millis{1000}; ///< probability thousands. one word.
+  std::atomic<unsigned int> promote_probability_millis{1000}; ///< probability thousands. one word.
```
Comment (tchaikov) [inline src/osd/OSD.h]: what's the rationale behind this change?

Comment (tchaikov) [reply src/osd/OSD.h]: @shinobu-x ping?

Comment (tchaikov) [reply src/osd/OSD.h]: @chardan i don't feel strong either way. i just don't think we want to add code churn for no reason.

Comment (chardan) [reply src/osd/OSD.h]: @shinobu-x Also, the title might need rewording, since this change /stops/ using typedef-ed specializations, the opposite of what it suggests.

**Comment by yuriw** [comment]
[wip-yuri4-testing-2018-01-02-1711](https://trello.com/c/NEW8iWlF/216-wip-yuri4-testing-2018-01-02-1711)

**Comment by chardan** [comment]
I'm still unclear on the purpose of this change.
